x-pack/plugins/infra/public/components/infrastructure_node_metrics_tables/shared/hooks/use_infrastructure_node_metrics.ts

```diff
@@ -109,10 +116,23 @@ export const useInfrastructureNodeMetrics = <T>(
 
   const pageCount = useMemo(() => Math.ceil(top100Nodes.length / TABLE_PAGE_SIZE), [top100Nodes]);
 
+  const data = useMemo<NodeMetricsTableData<T>>(
+    () =>
+      errors.length > 0
+        ? { state: 'error', errors }
+        : metricIndicesExist == null
+        ? { state: 'unknown' }
+        : !metricIndicesExist
+        ? { state: 'no-indices' }
+        : nodes.length <= 0
+        ? { state: 'empty-indices' }
+        : { state: 'data', currentPageIndex, pageCount, rows: nodes },
```

Comment:
You're just trying to hurt me on purpose now aren't you? 😢

Reply:
I too would prefer `if` to be an expression instead of a statement. But we have to work with what ECMAScript allows. What would you do differently here?

Reply:
> I will settle for an TC39 proposal

I actually just searched for one too :joy:

Reply:
https://github.com/tc39/proposal-pattern-matching maybe? It's stage 1 but very nice.